### PR TITLE
[Not For Merge] use ground truth to test segmentor

### PR DESCRIPTION
**Pull request by YiwenShaoStephen**
use ground truth to test segmentor.

### CODE REVIEW

**Comment by danpovey** [comment]
I think you might have forgotten to "git add" the actual test file.  Or am I missing something?  Where is the test?

**Comment by YiwenShaoStephen** [comment]
You only need to use run.sh —stage 2 to run segmentation. The current segment.py only segments one image for now so I didn’t add other things here. I just used it as a quick test. But if you think adding a test script is necessary for our future work, I can write one accordingly.

File: egs/dsb2018/v1/local/train.py [hunk 1]
```diff
@@ -281,7 +281,8 @@ def sample(model, dataloader, outdir, core_config):
         torchvision.utils.save_image(
             class_pred[:, i:i + 1, :, :], '{0}/class_pred{1}.png'.format(outdir, i))
 
-    return img, class_pred, bound_pred
+    # return ground truth for test
+    return img, classification, bound
```
Comment (danpovey) [inline egs/dsb2018/v1/local/train.py]: oh I see, maybe this is the key line.

**Comment by hhadian** [comment]
The new version of Pytorch complains that we should "detach" a variable (from the computation graph I guess) before we can call numpy() on it.

**Comment by YiwenShaoStephen** [comment]
@detach() means class_pred[0] doesn't have gradients. Previously in PyTorch (0.3), tensor and variable are different objects, where variable contains both the tensor and its gradient. In pytorch 0.4, tensor and variable are combined and detach() is one of the way to make a tensor without gradient.

**Comment by danpovey** [comment]
Incidentally, I notice that you are using Variable() in the code, but
according to this
https://pytorch.org/docs/master/autograd.html
Variable() is now deprecated.

**Comment by danpovey** [comment]
It looks to me like maybe with modern PyTorch you should use
torch.Tensor(), with 'requires_grad=false' for inputs for which we don't
need gradients.

**Comment by YiwenShaoStephen** [comment]
Sorry, it's a mistake. I will check if there is any other place that I forget to make it fit with torch 0.4.

**Comment by YiwenShaoStephen** [comment]
@danpovey requires_grad=false is deprecated in torch 0.4. Now the suggested way is detach() or with torch.no_grad: (maybe the syntax is not perfectly right).

**Comment by YiwenShaoStephen** [comment]
@danpovey Oh, sorry. I check it and find requires_grad=false is still in use but now for Tensor object. Previously it was in Variable so I thought it was removed simultaneously. Then requires_grad=false is still a recommendable way.
